infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/receive/LogReplicationSinkManager.java

```diff
@@ -398,14 +401,17 @@ private void processSnapshotMessage(LogReplicationEntry message) {
         switch (message.getMetadata().getMessageMetadataType()) {
             case SNAPSHOT_MESSAGE:
                 snapshotWriter.apply(message);
-                return;
+                break;
             case SNAPSHOT_END:
+                dataConsistent.set(false);
+                logReplicationMetadataManager.setDataConsistentOnStandby(dataConsistent.get());
                 snapshotWriter.snapshotTransferDone(message);
                 completeSnapshotApply(message);
-                return;
+                dataConsistent.set(true);
```

Comment:
To keep a sense of atomicity maybe we can have a private method setDataConsistency(boolean consistent) something like that where we update the local variable and the persisted...

Reply:
sure